File: backend/internal/types/RegistrationRequest.go
package types

import "fmt"

type RegistrationRequest struct {
	FirstName       string `json:"first_name"`
	LastName        string `json:"last_name"`
	Email           string `json:"email"`
	PhoneNo         string `json:"phone_no"`
	Fingerprint     string `json:"fingerprint"`
	ProofOfIdentity string `json:"proof_of_identity"`
	Location        string `json:"location"`
}

func (req RegistrationRequest) Validate() error {
	if req.FirstName == "" {
		return fmt.Errorf("first name is empty")
	}

	if req.LastName == "" {
		return fmt.Errorf("last name is empty")
	}

	if req.Email == "" {
		return fmt.Errorf("email is empty")
	}

	if req.PhoneNo == "" {
		return fmt.Errorf("phone number is empty")
	}

	if req.Fingerprint == "" {
		return fmt.Errorf("fingerprint is empty")
	}

	if req.ProofOfIdentity == "" {
		return fmt.Errorf("proof of identity is empty")
	}

	if req.Location == "" {
		return fmt.Errorf("location is empty")
	}

	return nil
}

File: frontend/src/lib/types/registrationRequest.ts
import { NewError } from "./CommonError"

export type RegistrationRequest = {
    first_name: string | undefined,
    last_name: string | undefined,
    email: string | undefined,
    phone_no: string | undefined,
    proof_of_identity: string | undefined,
    fingerprint: string | undefined,
    location: string | undefined
}

export const validateRegistrationRequest = function(req: RegistrationRequest) {
    console.log("request: ", req)
    if (req.email === undefined 
    || req.first_name === undefined
    || req.last_name === undefined
    || req.email === undefined
    || req.phone_no === undefined
    || req.proof_of_identity === undefined
    || req.fingerprint === undefined
    || req.location === undefined) {
        return NewError("missing data")
    }

    if (req.proof_of_identity === "") {
        return NewError("missing identity of proof")
    }

    if (req.fingerprint === "") {
        return NewError("missing fingerprint")
    }
}